source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+			return text.trim();
+		},
+
+		/**
+		* @private
+		*/
+		tick: function () {
+			var hr = this.valueArray['hour'];
+			var mn = this.valueArray['minute'];
+			var ss = this.valueArray['second'];
+			if (hr <= 0 && mn <= 0 && ss <= 0) {
+				this.pickerExpired();
+				return;
+			} else {
+				if (ss === 0 ) {
+					ss = 59;
+					this.valueArray['second'] = ss;
+					if (mn > 0 ) {
+						mn--;
+						this.valueArray['minute'] = mn;
+					} else {
+						mn = 59;
+						hr--;
+						this.valueArray['minute'] = mn;
+						this.valueArray['hour'] = hr;
+					}
+				} else {
+					ss--;
+					this.valueArray['second'] = ss;
+				}
+				this.set('value', this.formatValue());
+			}
+		},
+
+		/**
+		* @private
+		*/
+		updateValue: function (sender, ev) {
+			var hour = this.$.hourPicker ? this.$.hour.get('value') : this.valueArray['hour'];
+			var minute = this.$.minutePicker ? this.$.minute.get('value') : this.valueArray['minute'];
+			var second = this.$.secondPicker ? this.$.second.get('value') : this.valueArray['second'];
+			if (ev.name == 'second' && this.valueArray['second'] == second) {
+				return;
+			} else if (ev.name == 'minute' &&  this.valueArray['minute'] == minute) {
+				return;
+			} else if (ev.name == 'hour' && this.valueArray['hour'] == hour) {
+				return;
+			}
+			this.valueArray['second'] = second;
+			this.valueArray['minute'] = minute;
+			this.valueArray['hour'] = hour;
+			this.set('value', this.formatValue());
+		},
+
+		/**
+		* @private
+		*/
+		pickerExpired: function () {
+			this.cancelPicker();
+			this.doPickerExpired({name:this.name});
+		},
+
+		/**
+		* @private
+		*/
+		changeCountdown: function () {
+			if (this.countdown) {
+				if (this.$.hour)
+					this.$.hour.set('disabled', true);
+				if (this.$.minute)
+					this.$.minute.set('disabled', true);
+				if (this.$.second)
+					this.$.second.set('disabled' , true);
+			} else {
+				if (this.$.hour)
+					this.$.hour.set('disabled', false);
+				if (this.$.minute)
+					this.$.minute.set('disabled', false);
+				if (this.$.second)
+					this.$.second.set('disabled' , false);
+			}
+		},
+
+		/**
+		* @private
+		*/
+		startTapped: function () {
+			if (this.startConent == moon.$L('Start')) {
+				if(this.valueArray['hour'] || this.valueArray['minute'] || this.valueArray['second']) {
+					this.startPicker();
+				}
+			} else {
+				this.set('value', '');
+			}
+		},
+
+		/**
+		* @private
+		*/
+		pauseTapped: function () {
+			if (this.pauseConent == moon.$L('Pause')) {
+				this.set('pauseConent', moon.$L('Resume'));
+				this.pauseTimer();
+			} else {
+				this.set('pauseConent', moon.$L('Pause'));
+				this.startTimer();
+			}
+		},
+
+		/**
+		* @private
+		*/
+		startPicker: function () {
+			this.set('startConent', moon.$L('Cancel'));
+			this.set('pauseDisabled', false);
+			this.startTimer();
+		},
+
+		/**
+		* @private
+		*/
+		cancelPicker: function () {
+			this.set('startConent', moon.$L('Start'));
```

Comment:
In fact, we might want to consider storing these strings in a single location, possibly as private properties of this control, so we can update them all in one place. As it stands now, we would have to remember to update the text content for "Pause" in at least 4 locations, if we ever wanted to change it.